config/checker/git4idea.astub

```diff
@@ -12,7 +12,7 @@ class GitVcs {
 package git4idea.branch;
 
 class GitNewBranchDialog {
-  GitNewBranchDialog(Project project, Collection repositories, @Untainted String dialogTitle, @Untainted String initialName,
+  GitNewBranchDialog(Project project, Collection repositories, @Untainted String dialogTitle, String initialName,
```

Comment:
So it accepts HTML? 🤔

Reply:
@PawelLipski, I have a string `someString` from user input - ofc we do not know if it is tainted or untainted. how do I force its type to untainted?

Reply:
Still, I wouldn't change the stub. This place in git4idea only allows for non-HTML string, our stubs shouldn't indicate otherwise :/ 

> from user input

Hmmm actually... if user passes sth like `<b>foo</b>` as this input, what happens? 🤔 we should sanitize the input (which we probably already do... but good to check 😅) and reflect this sanitization in type system (some method that accepts `@Tainted` input and returns `@Untained`?)

Reply:
that's my problem about this (un)tainted thing. I think we overuse it in user input cases. and for some reason kotlin allows to pass anything into @Untainted (see com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt:75). 

yes. I think we need some mechanism that takes `@Tainted` and return `@Untainted` - I am goona to take care of it on a separate PR

Reply:
the situation is even worser - `<b>foo</b>` is a valid branch name!

Reply:
>  and for some reason kotlin allows to pass anything into `@Untainted`

The only way to enforce this taintedness is via Checker Framework, which only works for Java :/ this is wider problem, also affects checkers like UI Effect or I18n. That's the primary reason why I'm reluctant to switch to Kotlin even in new code... Java + Checker is safer than Kotlin. Even Nullness Checker is more powerful than Kotlin's built-in nullness tracking (esp. around constructors).

Rel: https://discuss.kotlinlang.org/t/potential-for-something-like-checker-framework-in-kotlin/1458

Reply:
>  I think we overuse it in user input cases

Whoops quite likely 😅 although there aren't many... just Checkout Branch action AFAICS

Reply:
ok, so in this topic it's a separate issue (#1242)